### PR TITLE
VxDesign: End-to-end prototype

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Phase 2 of the VxDesign project - an end-to-end prototype of the ballot layout workflow:
- Selecting an election definition
- Proofing the ballot (created by auto-layout)
- Exporting the ballot PDFs and ballot definition

_Review by commits_

Note that this is still experimental prototype code, so there are no automated tests. I still expect the product features to change rapidly, so I'm treating this code as temporary.

Feedback on overall file structure and application architecture is welcome, as well as anything you notice at a high level about the code. Product ideas/feedback welcome as well.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/530106/3437edd5-daec-44f9-a087-ffe004e8aed4

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/design/frontend/src/file_input_button.tsx
Comment (jonahkagan) [inline apps/design/frontend/src/file_input_button.tsx:1]: Copied from VxAdmin

**Comment by arsalansufi** [comment]
Sorry I haven't gotten to this yet! Will definitely have it reviewed before the start of your workday tomorrow

**Comment by jonahkagan** [comment]
No prob, I'm not blocked!

**Review by arsalansufi — APPROVED** [review]
This looks awesome!

File: apps/design/shared/package.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  "name": "@votingworks/design-shared",
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/design/shared/package.json:2]: Good points, thanks!

File: apps/design/shared/src/layout.ts [hunk 2]
```diff
@@ -0,0 +1,862 @@
...
+export const DOCUMENT_WIDTH = 8.5 * PPI;
+export const DOCUMENT_HEIGHT = 11 * PPI;
+
+const NUM_CONTEST_COLUMNS = 3;
+const GUTTER_WIDTH = 1;
+const CONTEST_COLUMN_WIDTH = 10;
+export const GRID: GridDimensions = {
+  rows: 41,
+  columns:
+    3 + // Timing marks
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/design/shared/src/layout.ts:89]: Good question, I can't remember!

I'm going back to a hardcoded number of columns in an upcoming PR so I'm not going to bother to figure it out right now.

File: apps/design/backend/src/app.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    getElection(): ElectionDefinition | null {
+      return store.getElection() ?? null;
+    },
+
+    setElection(input: { electionData?: string }): Result<void, Error> {
+      if (input.electionData) {
+        const parseResult = safeParseElection(input.electionData);
+        if (parseResult.isErr()) return parseResult;
+      }
+      store.setElection(input.electionData);
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/design/backend/src/app.ts]: Updated cabb9d307

File: apps/design/frontend/src/app.tsx [hunk 4]
```diff
@@ -1,301 +1,340 @@
...
+        >
+          Select Election
+        </FileInputButton>
+      </Main>
+    </Screen>
   );
 }
 
-const PAGE_GAP = 100;
+function HomeScreen(): JSX.Element | null {
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/design/frontend/src/app.tsx:155]: Agreed! Will keep that in mind as I flesh out the next phase

**Pull request merged**